main.cpp

```diff
@@ -531,6 +736,53 @@ int Arpeggiator::playNoteDuration(int noteDuration)
     return noteDuration;
 }
 
+void Arpeggiator::userSelectedNoteDuration(double noteDuration)
+{
+    std::map<double, std::string> noteLengths
+            {
+                    {4.0, "semibreve"},
+                    {2.0, "minim"},
+                    {1.0, "crotchet"},
+                    {0.5, "quaver"},
+                    {0.25, "semiquaver"}
+            };
+
+    std::vector<double> noteLengthVector = {4.0, 2.0, 1.0, 0.5, 0.25};
+    std::cout << "Chose from the following note durations: ";
+    for (auto c : noteLengthVector)
+        std::cout << c << ", ";
+
+    std::cout << std::endl;
+    {
+        if(noteDuration == 4.0)
+        {
+            std::cout << noteLengths[4.0] << " beats per 4-4 bar = 1";
+        }
+        else if(noteDuration == 2.0)
+        {
+            std::cout << noteLengths[2.0] << " beats per 4-4 bar = 2";
+        }
+        else if(noteDuration == 1.0)
+        {
+            std::cout << noteLengths[1.0] << " beats per 4-4 bar = 4";
+        }
+        else if(noteDuration == 0.5)
+        {
+            std::cout << noteLengths[0.5] << " beats per 4-4 bar = 8";
+        }
+        else if(noteDuration == 0.25)
+        {
+            std::cout << noteLengths[0.25] << " beats per 4-4 bar = 16";
+        }
+        else
+        {
+            std::cout << "ERROR! Note duration not available" << std::endl;
+        }
+    }
```

Comment:
Why is this chunk of code inside these `{ }`?
- [ ] revise?

Reply:
Nothing was revised.
Are you trying to print out this stuff inside the `for()` loop?
Why have you put all of these `if else if` statements inside a pair of `{ }`?
- [ ] revise!!